src/transformState.js

```diff
@@ -5,7 +5,32 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    const { type, keysToRemove, extraData } = action;
+
+    switch (type) {
+      case 'addProperties':
+        Object.assign(state, extraData);
+        break;
+
+      case 'removeProperties':
+        for (const keyToRemove of keysToRemove) {
+          delete state[keyToRemove];
+        }
+        break;
+
+      case 'clear':
+        for (const keyToRemoveInState in state) {
+          delete state[keyToRemoveInState];
+        }
+        break;
+
+      default:
+        return `Unknown type ${type} `;
+    }
+  }
+
+  return state;
```

Comment:
No need to return the state object at the end of the function. The function should modify the state object directly without returning a value.